py/dml/c_backend.py

```diff
@@ -1870,23 +1855,23 @@ def generate_init_data_objs(device):
             # mainly meant to capture EIDXVAR; for other errors, the error will
             # normally re-appear when evaluating per instance
             except DMLError:
-                for indices in node.all_indices():
-                    index_exprs = tuple(mkIntegerLiteral(node.site, i)
-                                        for i in indices)
-                    nref = mkNodeRef(node.site, node, index_exprs)
-                    try:
-                        init = eval_initializer(
-                            node.site, node._type, node.astinit,
-                            Location(node.parent, index_exprs), global_scope,
-                            True)
-                    except DMLError as e:
-                        report(e)
-                    else:
-                        if deep_const(node._type):
-                            coverity_marker('store_writes_const_field',
-                                            'FALSE',
-                                            node.site)
-                        init.assign_to(nref, node._type)
+                with allow_linemarks():
+                    for indices in node.all_indices():
+                        index_exprs = tuple(mkIntegerLiteral(node.site, i)
+                                            for i in indices)
+                        nref = mkNodeRef(node.site, node, index_exprs)
+                        try:
+                            init = eval_initializer(
+                                node.site, node._type, node.astinit,
+                                Location(node.parent, index_exprs),
+                                global_scope, True)
+                        except DMLError as e:
+                            report(e)
+                        else:
+                            markers = ([('store_writes_const_field', 'FALSE')]
+                                       if deep_const(node._type) else [])
+                            coverity_markers(markers, node.site)
+                            init.assign_to(nref, node._type)
```

Comment:
why linemark initializers? they are not interesting while debugging, and just add noise to coverage.

If you need to enable linemarks just for coverity, then I suppose we can change `coverity_markers` to go an easy path if linemarks are disabled?

Reply:
1. I don't need linemarks for coverity support; and 2. with this PR `coverity_markers` has such a path.

I could just as easily remove the `allow_linemarkers` and this will work. Like I said in R.C., linemarking initializers is a question if we want it for _other_ analysis tools than gcov or gdb. I mentioned Coverity as an example because if coverity finds issue with a session/saved initializer, then we would rather it point to the DML declaration, wouldn't we?